runelite-client/src/main/java/net/runelite/client/plugins/thieving/ChestOverlay.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2019, whs
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.thieving;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import net.runelite.api.Client;
+import net.runelite.api.Perspective;
+import net.runelite.api.Point;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.ProgressPieComponent;
+
+@Singleton
+class ChestOverlay extends Overlay
+{
+	private final Client client;
+	private final ThievingPlugin plugin;
+	private final ThievingConfig config;
+
+	@Inject
+	private ChestOverlay(final Client client, final ThievingPlugin plugin, final ThievingConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		this.plugin = plugin;
+		this.client = client;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		List<ChestRespawn> respawns = plugin.getRespawns();
+		if (respawns.isEmpty())
+		{
+			return null;
+		}
+
+		Instant now = Instant.now();
+		for (Iterator<ChestRespawn> it = respawns.iterator(); it.hasNext(); )
+		{
+			Color pieFillColor = config.respawnColor();
```

Comment:
It would be better to set the color on config changed events, rather than fetching it every frame

Reply:
Updated

Reply:
While it's close to what I meant, I meant subscribing in the plugin, and setting the field from there. Overlays don't have a "shutdown" event where you can unregister subscriptions. Convention-ish is keeping eventbus subscriptions all in the plugin class.

Also, it would be better to only change the fields when the config options for them have changed, like ```private void onConfigChanged(ConfigChanged event) {
   if (!"thieving".equals(event.getGroup()) {
      return;
   }

   switch (event.getKey()) {
      case "respawnColor":
         overlay.setRespawnColor(config.respawnColor());
         return;
      case ...